### ISSUE TITLE
Add files for "Show lock symbol for restricted items to all users" in vtechworks6-dev

### ISSUE THREAD

**Issue by alawvt** [VTUL/vtechworks#571]
Please add the files from PR, https://github.com/VTUL/vtechworks/pull/87/files, to dspace_6_dev so the lock symbol for restricted files will appear for all users.

**Comment by keithgee** [comment]
I'll look at this again soon. Adding the files from this previous pull request won't work with DSpace 6 for these reasons:

- The DSpace API has changed from version 5 to version 6 such that the files won't compile without modification to account for these changes.
- The customized "AuthorizeManager.java" file  serves as an overlay for a file in the DSpace release, but that original "AuthorizeManager.java" file has been removed in DSpace 6, thus the overlay will not customize anything, even after the compile errors have been fixed.

Also, @alawvt, can you recommend an item for testing? Previously, I think we tested with the famous "flotation of fines" item, but by looking at production it seems that item is no longer restricted.

**Comment by alawvt** [comment]
@keithgee, thank you for researching this issue. I hope that it will be manageable with the DSpace 6 API. 

Most of the [Creative Writing theses](https://vtechworks.lib.vt.edu/handle/10919/9291/discover?field=department&filtertype=department&filter_relational_operator=equals&filter=Creative+Writing) are restricted to campus only. Examples are https://vtechworks.lib.vt.edu/handle/10919/42646, https://vtechworks.lib.vt.edu/handle/10919/42110, and https://vtechworks.lib.vt.edu/handle/10919/42648.

**Comment by alawvt** [comment]
Two files that aren't in the dspace/modules folder are

- additions/src/main/java/org/dspace/authorize/AuthorizeManager.java # used for lock symbol commit, https://github.com/VTUL/vtechworks/commit/9b6d9625b390e9abc34dd4bac08ca2b775329a85, however dspace-api/src/main/java/org/dspace/authorize/AuthorizeManager.java **does not exist** in DSpace 6.

- xmlui/src/main/java/org/dspace/app/xmlui/objectmanager/ItemAdapter.java # used for lock symbol commit, https://github.com/VTUL/vtechworks/commit/9b6d9625b390e9abc34dd4bac08ca2b775329a85, dspace-xmlui/src/main/java/org/dspace/app/xmlui/objectmanager/ItemAdapter.java **does exist** in DSpace 6.

**Comment by alawvt** [comment]
@keithgee, I am running through some tests on the LDE with the latest build. The item, http://192.168.60.4/handle/10919/37514, has the lock symbols. I am trying to figure if it thinks I am off campus. What do you think I should see?

**Comment by keithgee** [comment]
@alawvt, it probably does think that you are off-campus. I notice that you took out all of the on-campus IP address ranges with commit 630fd2c0062f9223cfa390764e329b311ed8d914, which was a revert of a previous commit.

We'd recently talked about how to organize the ip-address configuration. I'm wondering if maybe you meant to only revert part of the previous commit, and leave the ip addresses in modules/authentification.cfg?

Basically, I think you accidentally removed the list of on-campus IP addresses from the configuration.

Does that check out?

**Comment by alawvt** [comment]
@keithgee, yes, that is exactly what happened. I reverted my commit and added another commit to just change authentication.cfg back to the DSpace6 default. The lock symbols now appear. Once again, thank you for figuring these things out!

**Comment by keithgee** [comment]
That's one thing sorted out, but I'm still confused about when locks should show/when they shouldn't and how to make it work again. :)

**Comment by alawvt** [comment]
@keithgee , If a bitstream is restricted, our customization is that it should display a lock symbol for all users, whether or not they are on campus.
